Create release with softprops/action-gh-release.

paul121 opened the pull request:
Our Github `release` workflow uses two github actions that were archived in March:

- https://github.com/actions/create-release
- https://github.com/actions/upload-release-asset

https://github.com/softprops/action-gh-release was recommended as a community maintained action. It seemed pretty popular, up to date, and was easy to integrate. Its default values match what we were using previously.

paul121 commented:
Tested in this branch: https://github.com/paul121/farmOS-map/commits/2.x-update-release-artifacts-test

Pushing the `v2.0.0-paul.0` tag created this release: https://github.com/paul121/farmOS-map/releases/tag/v2.0.0-paul.0

The changelog link doesn't work because the `farmOS/farmOS-map` repository is hard coded in the release body.

mstenta commented:
Wow bummer that the official GitHub repos are no longer maintained. Seems weird that they wouldn't provide support for this, from a trust/security perspective. I'm sure https://github.com/softprops/action-gh-release is trustworthy, but I can't help but feel like outsourcing to a community project for packaging is a security risk - maybe not as much for something like farmOS, but recent history (eg: [event-stream in NPM](https://blog.npmjs.org/post/180565383195/details-about-the-event-stream-incident)) shows that it can happen.

I guess we'll have to consider a similar change in farmOS itself too: https://github.com/farmOS/farmOS/blob/1e3c018fe0e17a76d1757830ea3f3e0dbab11d1a/.github/workflows/package-release.yml#L27

paul121 commented:
Ah dang. That is a good consideration. It might be possible to pin to a more specific version (the examples just show `@v1`)?

I briefly reviewed all the actions mentioned in the `actions/create-release` readme. This one seemed simplest but maybe it is worth further review if we will be using this elsewhere

mstenta commented:
> I briefly reviewed all the actions mentioned in the `actions/create-release` readme. This one seemed simplest but maybe it is worth further review if we will be using this elsewhere

Yea this one looks to have the most momentum (just going by GitHub stars/forks). So I agree this one is probably the best to start with.

I'm not too concerned about the security implications in our case - just felt the need to voice my disappointment that GitHub itself isn't supporting an "official" action for this anymore.

I'm OK with merging this. :+1:

symbioquine reviewed (approved):
LGTM - other than the version pinning question...

> Ah dang. That is a good consideration. It might be possible to pin to a more specific version (the examples just show `@v1`)?

I think you could do;

```yaml
  uses: softprops/action-gh-release@2d72d869af3bf23602f9593a1e3fd739b80ac1eb #v0.1.12
```

Ref:

* https://github.com/softprops/action-gh-release/releases
* https://docs.github.com/en/actions/learn-github-actions/security-hardening-for-github-actions#using-third-party-actions
* https://michaelheap.com/ensure-github-actions-pinned-sha/

mstenta commented:
I like that approach. And GitHub recommends it in the security hardening docs, so I guess that's the best way. :-)

mstenta commented:
So I think we just want to tweak this slightly to use the commit hash, yes? Then we can merge it?

paul121 commented:
Sorry for the delay!!

Rebased, pinned to `6034af24fba4e5a8e975aaa6056554efe4c794d0` [v0.1.13](https://github.com/softprops/action-gh-release/releases/tag/v0.1.13), and updated CHANGELOG.md.
